debugger/src/rtt.rs

```diff
@@ -0,0 +1,388 @@
+    #[structopt(skip)]
+    #[serde(default, rename = "rtt_enabled")]
+    pub enabled: bool,
+    /// Connection timeout in ms.
+    #[structopt(skip)]
+    #[serde(default = "default_rtt_timeout", rename = "rtt_timeout")]
+    pub timeout: usize,
+    /// Configure data_format and show_timestamps for select channels
+    #[structopt(skip)]
+    #[serde(default = "default_channel_formats", rename = "rtt_channel_formats")]
+    pub channels: Vec<RttChannelConfig>,
+}
+
+/// The User specified configuration for each active RTT Channel. The configuration is passed via a DAP Client configuration (`launch.json`). If no configuration is specified, the defaults will be Dataformat::String and show_timestamps=false
+#[derive(StructOpt, Debug, Clone, serde::Deserialize, Default)]
+pub struct RttChannelConfig {
+    pub channel_number: Option<usize>,
+    pub channel_name: Option<String>,
+    #[serde(default)]
+    pub data_format: DataFormat,
+    #[structopt(skip)]
+    #[serde(default)]
+    pub show_timestamps: bool,
+}
+
+/// This is the primary interface through which RTT channel data is read and written. Every actual RTT channel has a configuration and buffer that is used for this purpose.
+#[derive(Debug)]
+pub struct RttActiveChannel {
+    pub up_channel: Option<UpChannel>,
+    pub down_channel: Option<DownChannel>,
+    pub channel_name: String,
+    pub data_format: DataFormat,
+    /// Data that will be written to the down_channel (host to target)
+    input_data: String,
+    rtt_buffer: RttBuffer,
+    show_timestamps: bool,
+}
+
+/// A fully configured RttActiveChannel. The configuration will always try to 'default' based on information read from the RTT control block in the binary. Where insufficient information is available, it will use the supplied configuration, with final hardcoded defaults where no other information was available.
+impl RttActiveChannel {
+    pub fn new(
+        up_channel: Option<UpChannel>,
+        down_channel: Option<DownChannel>,
+        channel_config: Option<RttChannelConfig>,
+    ) -> Self {
+        let full_config = match channel_config {
+            Some(channel_config) => channel_config,
+            None => RttChannelConfig {
+                ..Default::default() // Will set intelligent defaults below ...
+            },
+        };
+        let name = up_channel
+            .as_ref()
+            .and_then(|up| up.name().map(Into::into))
+            .or_else(|| {
+                down_channel
+                    .as_ref()
+                    .and_then(|down| down.name().map(Into::into))
+            })
+            .or_else(|| full_config.clone().channel_name)
+            .unwrap_or_else(|| "Unnamed RTT channel".to_string());
+        let buffer_size: usize = up_channel
+            .as_ref()
+            .map(|up| up.buffer_size())
+            .or_else(|| down_channel.as_ref().map(|down| down.buffer_size()))
+            .unwrap_or(1024); // This should never be the case ...
```

Comment:
If this can't ever be the case, please make it an expect which explains what was expected and to report this bug (because it shouldn't be hit).
E.g. `.expect("Channel buffer size is unknown. This should never happen and should be reported as a bug.")`

Reply:
My comment was misleading ... I've updated it to be less dramatic LOL